src/pudl/settings.py

```diff
@@ -348,6 +348,88 @@ class GlueSettings(FrozenBaseModel):
     ferc1: bool = True
 
 
+@unique
+class GPRATKTechType(StrEnum):
+    """Enum to constrain GridPath RA Toolkit technology types."""
+
+    WIND = auto()
+    SOLAR = auto()
+    # Not yet implemented
+    # THERMAL = auto()
+
+
+@unique
+class GPRATKProcLevel(StrEnum):
+    """Enum to constraint GridPath RA Toolkit processing levels."""
+
+    EXTENDED = auto()
+    # Not yet implemented
+    # AGGREGATED = auto()
+    # ORIGINAL = auto()
+
+
+class GridPathRAToolkitSettings(GenericDatasetSettings):
+    """An immutable pydantic model to validate GridPath RA Toolkit settings."""
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("gridpathratoolkit")
+    technology_types: list[str] = ["wind", "solar"]
+    processing_levels: list[str] = ["extended"]
+    daily_weather: bool = True
+    parts: list[str] = []
```

Comment:
I think our default here across the other setting classes is to include all of the working partitions.

Reply:
I guess that's true.